Apps/phvisualping/visualping_connector.py

```diff
@@ -0,0 +1,428 @@
+
+        # make rest call
+        ret_val, response = self._make_rest_call('/api/job/list', action_result, params=None, headers=None)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        # Add the response into the data section
+        action_result.add_data(response)
+        try:
+            jobs = response['jobs']
+            active_jobs = jobs['active']
+        except Exception as e:
+            error_message = self._get_error_message_from_exception(e)
+            self.debug_print("Error message: {}".format(error_message))
+            return action_result.set_status(phantom.APP_ERROR, UNEXPECTED_RESPONSE_MSG)
+
+        # Add a dictionary that is made up of the most important values from data into the summary
+        summary = action_result.update_summary({})
+        summary['active_jobs'] = active_jobs
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_show_result(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        id = param['id']
+
+        # make rest call
+        # for some reason the /api/job/lastresult returns a ContentType: text/html json response
+        ret_val, response = self._make_rest_call('/api/job/lastresult/{0}'.format(id), action_result, params=None, headers=None)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        # Add the response into the data section
+        action_result.add_data(response)
+        # Add a dictionary that is made up of the most important values from data into the summary
+        summary = action_result.update_summary({})
+
+        try:
+            summary['url'] = response['url']
+            summary['description'] = response['description']
+            summary['lastCheck'] = response['lastcheck']
+            summary['percentDifference'] = response['PercentDifference']
+        except Exception as e:
+            error_message = self._get_error_message_from_exception(e)
+            self.debug_print("Error message: {}".format(error_message))
+            return action_result.set_status(phantom.APP_ERROR, UNEXPECTED_RESPONSE_MSG)
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_get_images(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        id = param['id']
+
+        # make rest call
+        ret_val, response = self._make_rest_call('/api/job/lastimage/{0}'.format(id), action_result, params=None, headers=None)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        # Add the response into the data section
+        action_result.add_data(response)
```

Comment:
Are the images supposed to be uploaded to the vault/file folder?

Reply:
No